tests/test_views_list_post.py

```diff
@@ -140,3 +140,29 @@ def test_validation_error(app, db, test_data, search_url, content_type):
         res = client.post(
             search_url, data=json.dumps(test_data[0]), headers=HEADERS)
         assert res.status_code == 400
+
+
+@mock.patch('invenio_records.api.Record.validate', _mock_validate_fail)
+@pytest.mark.parametrize('content_type', [
+    'application/json', 'application/json;charset=utf-8'
+])
+def test_jsonschema_validation_error(app, db, search_url, content_type):
+    """Test when jsonschema validation fails."""
+    record = {
+        'title': 1,
+        '$schema': {
+            'properties': {
+                'title': {'type': 'string'}
+            }
+        }
+    }
+    with app.test_client() as client:
+        HEADERS = [
+            ('Accept', 'application/json'),
+            ('Content-Type', content_type)
+        ]
+
+        # Create record
+        res = client.post(
+            search_url, data=json.dumps(record), headers=HEADERS)
+        assert res.status_code == 400
```

Comment:
Specifically in this case, I would also test the content of the response, since this line will pass with the old exception class as well. I.e. something a long the lines of :

```assert res.json['message']```